.github/workflows/publish.yml

```diff
@@ -0,0 +1,121 @@
+name: Publish Docker image
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+permissions:
+  contents: read
+
+jobs:
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+        options: --health-cmd="pg_isready" --health-interval=10s --health-timeout=5s --health-retries=3
+
+    steps:
+      - name: Install packages
+        run: sudo apt-get update && sudo apt-get install --no-install-recommends -y google-chrome-stable curl libvips postgresql-client libpq-dev
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: .ruby-version
+          bundler-cache: true
+
+      - name: Run tests and smoke test seed
+        env:
+          RAILS_ENV: test
+          DATABASE_URL: postgres://postgres:postgres@localhost:5432
+        run: |
+          bin/rails db:create
+          bin/rails db:schema:load
+          bin/rails test:all
+          bin/rails db:seed
+
+      - name: Keep screenshots from failed system tests
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: screenshots
+          path: ${{ github.workspace }}/tmp/screenshots
+          if-no-files-found: ignore
+
+  build:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    needs: [test]
+    permissions:
+      packages: write
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Context for Buildx
+        run: docker context create builders
+        shell: bash
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          endpoint: builders
+
+      - name: Log in to the container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          flavor: latest=auto
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}.{{minor}}.{{patch}}
+            type=ref,event=branch
+
+      - name: Build and push Docker image
```

Comment:
It would be nice to build docker image for arm64

Reply:
That's probably more a question to the core maintainers since that could bring some additional overhead :D

Reply:
It's working perfectly fine -> https://github.com/DorianMazur/maybe/actions/runs/8822599388/job/24221266397

Reply:
@DorianMazur couple questions:

1. What device do you need this architecture for?
2. Any way we could reduce the build time on this?

Reply:
@zachgoll Maybe we could reduce build time by using `macos-latest` runners (arm64 architecture). Because currently in my workflow I am using ubuntu-latest that is x86_64. 

I am working on arm64, I have multiple homeLab servers (each of them is using 5W of power) on arm64 and my Macbook is also arm64. That's why I need this, probably not only me :)

Reply:
Gotcha, had forgotten about the M1 chip and Docker compatibility.  I'm guessing we'll have a decent number of self-hosters running this on a Mac, so probably worth supporting.

That said, I think we either need to get that build time significantly reduced _or_ we could change the frequency in which we're running these builds.

I was originally thinking we'd build and publish on every commit to `main`, but it may be more practical to only build and publish for each new release (which we're eventually going to hit a weekly/bi-weekly cycle on).

Reply:
Not worried about cost—mostly want to be conscious of the overall feedback loop of making modifications to this flow in the future.

If we add this architecture, looks like we can add `runs-on: ${{ matrix.os }}` syntax ([reference](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-multi-dimension-matrix)) to speed things up.